lpwan-backend/src/main/java/com/cumulocity/lpwan/lns/connection/model/LpwanDeviceFilter.java

```diff
@@ -3,17 +3,30 @@
 import com.cumulocity.sdk.client.ParamSource;
 import com.cumulocity.sdk.client.inventory.InventoryFilter;
 
+import java.util.Map;
+import java.util.Optional;
+
 public class LpwanDeviceFilter extends InventoryFilter {
```

Comment:
You can simplify this class as following

```
public class LpwanDeviceFilter extends InventoryFilter {

    private static String FILTER_BY_SERVICE_PROVIDER_AND_LNS_CONNECTION_NAME = "$filter=(c8y_LpwanDevice.serviceProvider eq '%s' and c8y_LpwanDevice.lnsConnectionName eq '%s')";
    
    @ParamSource
    private String query;

    private LpwanDeviceFilter(String query) {
        this.query = query;
    }

    public static LpwanDeviceFilter byServiceProviderAndLnsConnectionName(String serviceProvider, String lnsConnectionName) {
        return new LpwanDeviceFilter(String.format(FILTER_BY_SERVICE_PROVIDER_AND_LNS_CONNECTION_NAME, encode(serviceProvider), encode(lnsConnectionName)));
    }
}
```

- [ ] Please note in the above code where I encode the values before placing it into the filter string. Like below, which is very important.
`encode(serviceProvider) and encode(lnsConnectionName)`

Reply:
The class has been simplified now